src/sentry/api/endpoints/project_rule_preview.py

```diff
@@ -39,13 +42,18 @@ def post(self, request: Request, project) -> Response:
             raise ValidationError
 
         data = serializer.validated_data
+        try:
+            endpoint = parse_datetime(request.data["endpoint"])
+        except (TypeError, ValueError, KeyError):
+            endpoint = timezone.now()
```

Comment:
Not sure if I should put this in the serializer, it's only needed for previews and none of the other rule serializers.